src/abstractdataframe/subset.jl

```diff
@@ -0,0 +1,231 @@
+
+If `skipmissing=true` returing `missing` in `args` is allowed and these rows are
+dropped. If `skipmissing=false` (the default) an error is thrown if `missing` is
+present.
+
+Each argument passed in `args` can be either a single column selector or a
+`source_columns => function` transformation specifier following the rules
+described for [`select`](@ref).
+
+Note that as opposed to [`filter!`](@ref) the `subset!` function works on whole
+columns (or all rows in groups for `GroupedDataFrame`) and by default drops rows
+for which contition is false.
+
+If `GroupedDataFrame` is subsetted then it must include all groups present in the
+`parent` data frame, like in [`select!`](@ref).
+
+See also: [`subset`](@ref), [`filter`](@ref), [`filter!`](@ref)
+
+# Examples
+
+```
+julia> df = DataFrame(id=1:4, x=[true, false, true, false], y=[true, true, false, false])
+4×3 DataFrame
+│ Row │ id    │ x    │ y    │
+│     │ Int64 │ Bool │ Bool │
+├─────┼───────┼──────┼──────┤
+│ 1   │ 1     │ 1    │ 1    │
+│ 2   │ 2     │ 0    │ 1    │
+│ 3   │ 3     │ 1    │ 0    │
+│ 4   │ 4     │ 0    │ 0    │
+
+julia> subset!(copy(df), :x, :y => ByRow(!));
+
+julia> df
+4×3 DataFrame
+│ Row │ id    │ x    │ y    │
+│     │ Int64 │ Bool │ Bool │
+├─────┼───────┼──────┼──────┤
+│ 1   │ 1     │ 1    │ 1    │
+│ 2   │ 2     │ 0    │ 1    │
+│ 3   │ 3     │ 1    │ 0    │
+│ 4   │ 4     │ 0    │ 0    │
+
+julia> df = DataFrame(id=1:4, y=[true, true, false, false], v=[1, 2, 11, 12]);
+
+julia> subset!(groupby(df, :y), :v => x -> x .> minimum(x));
+
+julia> df
+2×3 DataFrame
+│ Row │ id    │ y    │ v     │
+│     │ Int64 │ Bool │ Int64 │
+├─────┼───────┼──────┼───────┤
+│ 1   │ 2     │ 1    │ 2     │
+│ 2   │ 4     │ 0    │ 12    │
+
+julia> df = DataFrame(id=1:4, x=[true, false, true, false],
+                             z=[true, true, missing, missing], v=1:4);
+
+julia> subset!(df, :x, :z)
+ERROR: ArgumentError: each transformation must return a vector whose eltype is subtype of Bool
+
+julia> subset!(df, :x, :z, skipmissing=true);
+
+julia> df
+1×4 DataFrame
+│ Row │ id    │ x    │ z     │ v     │
+│     │ Int64 │ Bool │ Bool? │ Int64 │
+├─────┼───────┼──────┼───────┼───────┤
+│ 1   │ 1     │ 1    │ 1     │ 1     │
+
+julia> df = DataFrame(id=1:4, x=[true, false, true, false], y=[true, true, false, false],
+                             z=[true, true, missing, missing], v=[1, 2, 11, 12]);
+
+julia> subset!(groupby(df, :y), :v => x -> x .> minimum(x));
+
+julia> df
+2×5 DataFrame
+│ Row │ id    │ x    │ y    │ z       │ v     │
+│     │ Int64 │ Bool │ Bool │ Bool?   │ Int64 │
+├─────┼───────┼──────┼──────┼─────────┼───────┤
+│ 1   │ 2     │ 0    │ 1    │ 1       │ 2     │
+│ 2   │ 4     │ 0    │ 0    │ missing │ 12    │
+```
+"""
+function subset!(df::AbstractDataFrame, @nospecialize(args...); skipmissing::Bool=false)
+    row_selector = _get_subset_conditions(df, args, skipmissing)
+    return delete!(df, findall(!, row_selector))
```

Comment:
Doesn't `delete!` call `findall` automatically? Same below.

Reply:
it does, but `delete!` does the reverse operation (i.e. it requires passing rows to be deleted, and in `select` we select rows to be kept, so I have to negate the result before passing it to `delete!`)